source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -81,33 +81,38 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
   }
 
   initiateCall(headers);
-  return filter_return_ == FilterReturn::StopDecoding ? Http::FilterHeadersStatus::StopIteration
-                                                      : Http::FilterHeadersStatus::Continue;
+  return filter_return_ == FilterReturn::StopDecoding
+             ? Http::FilterHeadersStatus::StopAllIterationAndWatermark
```

Comment:
If I understood correctly, `StopAllIterationAndWatermark` will hold headers, data and trailers. It will also return an HTTP 413 if the buffer limit is exceeded. If that's correct, isn't a change of behaviour?

Reply:
Thanks for the review! StopAllIterationAndWatermark behaves the same as if StopIterationAndWater is returned by decodeData() (the original behavior), instead of returning 413, reading will be disabled until buffered data are processed. Does it make sense?

Reply:
No problem! Sorry, I was looking at the description for `StopAllIterationAndBuffer`. Thanks!